src/main/kotlin/com/example/ipwhitelist/repository/AppRepository.kt

```diff
@@ -0,0 +1,223 @@
+package com.example.ipwhitelist.repository
+
+import com.example.ipwhitelist.model.EnhancedAppResponse
+import com.example.ipwhitelist.model.dynamodb.Application
+import com.example.ipwhitelist.model.dynamodb.AppTableKeyPrefix
+import com.example.ipwhitelist.model.dynamodb.ApplicationDetails
+import com.example.ipwhitelist.model.dynamodb.ApplicationUser
+import org.springframework.stereotype.Repository
+import software.amazon.awssdk.enhanced.dynamodb.*
+import software.amazon.awssdk.enhanced.dynamodb.model.*
+import software.amazon.awssdk.enhanced.dynamodb.model.QueryConditional.sortBeginsWith
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue
+import java.util.*
+import kotlin.collections.ArrayList
+
+@Repository
+class AppRepository(
+    val dynamoDbEnhancedClient: DynamoDbEnhancedClient
+) {
+    fun <T : Application> save(application: T): T {
+        val applicationTable = getTable(application.javaClass)
+        applicationTable.putItem(application)
+        return application
+    }
+
+    fun getAllApps(): List<ApplicationDetails> {
+        val appTable = getTable(ApplicationDetails::class.java)
+        val items: MutableList<ApplicationDetails> = ArrayList()
+
+        val scanEnhancedRequest = ScanEnhancedRequest.builder()
+            .filterExpression(
+                Expression.builder()
+                    .expression("begins_with(objectId, :prefix)")
+                    .expressionValues(Collections.singletonMap(
+                        ":prefix", AttributeValue.builder()
+                            .s(AppTableKeyPrefix.APP.prefix).build()))
+                    .build()
+            )
+            .build()
+
+        val pageIterable: PageIterable<ApplicationDetails> = appTable.scan(scanEnhancedRequest)
+
+        for (page: Page<ApplicationDetails> in pageIterable) {
+            items.addAll(page.items())
+        }
+
+        return items
+    }
+
+    fun getAppsByUserId(userId: String) : List<EnhancedAppResponse> {
+
+        val scanEnhancedRequest = ScanEnhancedRequest.builder()
```

Comment:
I think that for getting apps by user id, we can use the appid GSI to query instead as scanning the table is quite expensive

Reply:
Good point, fixed